app/services/return-logs/setup/create-return-submission.service.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+/**
+ * Creates a new return submission, superseding a previous one if it exists
+ * @module CreateReturnSubmissionService
+ */
+
+const { generateUUID, timestampForPostgres } = require('../../../lib/general.lib.js')
+const ReturnSubmissionModel = require('../../../models/return-submission.model.js')
+
+/**
+ * Creates a new return submission. The version number will be 1 if no previous submission exists for this return log
+ * ID; otherwise, the version number will be incremented by 1
+ *
+ * If a previous submission does exist then it will be marked as superseded by setting its `current` property to false
+ *
+ * @param {string} returnLogId - ID of the return log (typically something like v1:6:01/23:987654:2024-11-01:2025-10-31)
+ * @param {string} userId - ID of the user creating the return submission (typically an email address)
+ * @param {object} metadata - Metadata for the return submission
+ * @param {boolean} nilReturn - Indicates if the return is a nil return
+ * @param {object} [trx=null] - Optional {@link https://vincit.github.io/objection.js/guide/transactions.html#transactions | transaction object}
+ *
+ * @returns {Promise<module:ReturnSubmissionModel>} - The created return submission
+ */
+async function go(returnLogId, userId, metadata, nilReturn, trx = null) {
```

Comment:
From just looking at this PR I'm not really sure why `trx = null` is there. Is it possible to do a unit test that includes a transaction just to clarify?

Reply:
Good shout -- I've added tests to his plus the other PR to verify that the `trx` is correctly used so if there are any errors then nothing is persisted in the db (which is why it's there -- the orchestrating service cals each of the services within a transaction so we don't end up with some stuff persisted and some not if it errors)